### PR TITLE
THREESCALE-10259: URL-encode data in the Download CSV link

### PR DESCRIPTION
**Pull request by mayorova**
**What this PR does / why we need it**:

Encode the data that is used for the CSV export of charts.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-10259

**Verification steps** 

1. Create an app with a special character, e.g. #, and create some traffic on it
2. Go to Analytics > Top Applications of the product, and click Download CSV
3. Verify that the CSV is generated correctly (no data is truncated)

**Special notes for your reviewer**:

### CODE REVIEW

**Review by jlledom — DISMISSED** [review]
Tried locally. Works fine 👍

File: app/javascript/src/Stats/lib/csv_link.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ export class StatsCSVLink extends StatsUI {
 
   buildCSVString (data) {
     let csvMatrix = this._parseData(data)
-    return csvMatrix.map(cell => cell.join(',')).join('%0A')
+    return csvMatrix.map(row => row.map(cell => encodeURIComponent(cell)).join(',')).join('%0A')
```
Comment (akostadinov) [inline app/javascript/src/Stats/lib/csv_link.js:29]: works locally, what I'm concerned though, how did you choose the `encodeURIComponent` function? This encoding stuff is a little tricky sometimes to get right

**Review by akostadinov — APPROVED** [review]

**Comment by akostadinov** [comment]
Works locally, so I approve, it is an improvement. I see another issue though. If app name contains a comma, then the CSV gets a broken header with more commas than anticipated.
[data3.csv](https://github.com/3scale/porta/files/13741664/data3.csv)

**Comment by akostadinov** [comment]
Actually it seems like in CSV to include commas in data, one should double quote the data field. `"app name, with comma"`, maybe add to thir PR or another one, as you like.

File: app/javascript/src/Stats/lib/csv_link.js (hunk 1 above)
Comment (mayorova) [reply app/javascript/src/Stats/lib/csv_link.js:29]: Well... This is a URL, so I thought that should work :sweat_smile:

**Comment by akostadinov** [comment]
But then the double quotes need to be handled if present in the name. Probably use some common library to escape fields or generate the whole CSV. It's not necessary to reinvent this.

**Comment by mayorova** [comment]
Good points, @akostadinov 
Except that I couldn't quickly find any library for that, and I don't think it's needed as it's just a couple of matches and replaces.

I added it here: https://github.com/3scale/porta/pull/3658/commits/54817dd48bada166b6bb4367ca84bdc0c0dfebcb

File: spec/javascripts/Stats/lib/csv_link.spec.js [hunk 2]
```diff
@@ -15,18 +15,28 @@ describe('StatsCSVLink', () => {
...
       ]
     ]
   }
-  const expectedCsvString = 'datetime,zaphod,marvin%0A11 Mar 1952 07:00:00 GMT,12,11%0A11 May 2001 08:00:00 BST,30,31'
+  const cells = [
+    ['datetime', '"with ""quotes"""', 'marvin #2', '"with, comma"'],
+    ['11 Mar 1952 07:00:00 GMT', '12', '11', '10'],
+    ['11 May 2001 08:00:00 BST', '30', '31', '32']
+  ]
+  const expectedCsvString = cells.map(row => row.map(cell => encodeURIComponent(cell)).join(',')).join('%0A')
```
Comment (mayorova) [inline spec/javascripts/Stats/lib/csv_link.spec.js:39]: This "expected string" assignment might be a bit ridiculous, as it's basically replicating what the app logic itself does... But the thing is that if we provide just the expected string, it is completely unreadable:
```
datetime,%22with%20%22%22quotes%22%22%22,marvin%20%232,%22with%2C%20comma%22%0A11%20Mar%201952%2007%3A00%3A00%20GMT,12,11,10%0A11%20May%202001%2008%3A00%3A00%20BST,30,31,32
```

**Review by akostadinov — APPROVED** [review]
Maybe not necessary to enclose in double quotes when the only special character is double quote but result should be valid and parseable anyway. Thank you!

**Comment by mayorova** [comment]
> Maybe not necessary to enclose in double quotes when the only special character is double quote but result should be valid and parseable anyway. Thank you!

Yeah, that was just easier, because otherwise we'd need to take care of two paths:
- quote, but NOT others - no change
- quotes or others - enclose and escape quotes

**Review by jlledom — APPROVED** [review]

**Pull request merged**